Src/orbtraceIf.c

```diff
@@ -472,8 +473,43 @@ bool OrbtraceGetIfandEP( struct OrbtraceIf *o )
             return false;
 
         case DEVICE_BMP: // -----------------------------------------------------------------------------
-            o->iface = BMP_IFACE;
-            o->ep = BMP_EP;
+            genericsReport( V_DEBUG, "Searching for BMP trace interface" EOL );
+
+            if ( ( err = libusb_get_active_config_descriptor( o->dev, &config ) ) < 0 )
+            {
+                genericsReport( V_WARN, "Failed to get config descriptor (%d)" EOL, err );
+                return false;
+            }
+
+            /* Loop through the interfaces looking for ours */
+            for ( int if_num = 0; if_num < config->bNumInterfaces && !interface_found; if_num++ )
+            {
+                const struct libusb_interface_descriptor *i = &config->interface[if_num].altsetting[0];
+
+                if (i->bInterfaceNumber != BMP_IFACE)
```

Comment:
Rather than assuming the interface number is always going to be the same, please instead search on the string "Black Magic Trace Capture" (`i->iInterface` to grab the appropriate string descriptor index) and validate there is a single IN endpoint. This will be the trace interface and will additionally allow Orbuculum to handle when the firmware it's looking at has had the trace interface omitted.

Reply:
Good point! Didnt think about ommited trace functionality. I will rework it in the evening.

Reply:
done